homeassistant/components/home_connect/api.py

```diff
@@ -348,31 +400,49 @@ class Oven(
     DeviceWithPrograms,
     DeviceWithRemoteControl,
     DeviceWithRemoteStart,
+    DeviceWithLocalControl,
+    DeviceWithProgramFinished,
 ):
     """Oven class."""
 
-    PROGRAMS = [
-        {"name": "Cooking.Oven.Program.HeatingMode.PreHeating"},
-        {"name": "Cooking.Oven.Program.HeatingMode.HotAir"},
-        {"name": "Cooking.Oven.Program.HeatingMode.TopBottomHeating"},
-        {"name": "Cooking.Oven.Program.HeatingMode.PizzaSetting"},
-        {"name": "Cooking.Oven.Program.Microwave.600Watt"},
-    ]
+    """Oven programs can't run by switches."""
 
     power_off_state = BSH_POWER_STANDBY
 
+    def get_oven_only_sensors(self):
+        """Get a dictionary with info about oven only sensors."""
+        return [
+            {
+                ATTR_DEVICE: self,
+                ATTR_DESC: "Temperature",
+                ATTR_KEY: COOKING_OVEN_TEMPERATURE,
+                ATTR_DEVICE_CLASS: DEVICE_CLASS_TEMPERATURE,
+                ATTR_UNIT: None,
+                ATTR_ICON: None,
+            }
+        ]
+
     def get_entity_info(self):
         """Get a dictionary with infos about the associated entities."""
         door_entity = self.get_door_entity()
         remote_control = self.get_remote_control()
         remote_start = self.get_remote_start()
+        local_control = self.get_local_control()
+        program_finished = self.get_program_finished()
         op_state_sensor = self.get_opstate_sensor()
         program_sensors = self.get_program_sensors()
+        oven_only_sensors = self.get_oven_only_sensors()
```

Comment:
Update to `get_oven_temperature_sensor`
```suggestion
        oven_temp_sensors = self.get_oven_temperature_sensors()
```